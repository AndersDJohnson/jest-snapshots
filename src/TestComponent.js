import React from 'react'
import PropTypes from 'prop-types'

const TestComponent = ({ one, two }) => <div>hello, {one}, {two}</div>

TestComponent.propTypes = {
  one: PropTypes.string.isRequired,
  two: PropTypes.string
}

export default TestComponent
