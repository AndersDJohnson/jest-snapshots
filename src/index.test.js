import { join } from 'path'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import snapshots from '.'
import TestComponent from './TestComponent'

configure({ adapter: new Adapter() });

const testComponentPath = join(__dirname, 'TestComponent')
const testComponentPathArray = [__dirname, 'TestComponent']

snapshots(testComponentPath)

snapshots(`${testComponentPath}.js`)

snapshots(testComponentPathArray)

snapshots({
  component: testComponentPath
})

snapshots({
  component: testComponentPath,
  mount: true
})

snapshots({
  component: testComponentPath,
  fakePropsOptions: {
    optional: true
  }
})

snapshots(TestComponent)

snapshots({
  component: TestComponent,
  props: [
    {
      one: 'A'
    }
  ]
})

snapshots(TestComponent, [
  {},
  {
    one: 'A'
  },
  {
    one: 'A',
    two: 'B',
    three: 'C'
  }
])

snapshots(testComponentPath, [
  {
    one: 'A'
  },
  {
    one: 'A',
    two: 'B',
    three: 'C'
  }
])

snapshots(testComponentPathArray, [
  {
    one: 'A'
  },
  {
    one: 'A',
    two: 'B',
    three: 'C'
  }
])

snapshots(TestComponent, {
  'with none': {},
  'does thing': {
    one: 'A'
  },
  'does things': {
    one: 'A',
    two: 'B',
    three: 'C'
  }
})

snapshots(testComponentPath, {
  'does thing': {
    one: 'A'
  },
  'does things': {
    one: 'A',
    two: 'B',
    three: 'C'
  }
})

snapshots(testComponentPathArray, {
  'does thing': {
    one: 'A'
  },
  'does things': {
    one: 'A',
    two: 'B',
    three: 'C'
  }
})
