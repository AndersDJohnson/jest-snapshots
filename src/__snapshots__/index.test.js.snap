// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent snapshots 0 1`] = `
<div>
  hello, 
  , 
</div>
`;

exports[`TestComponent snapshots 0 2`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots 0 3`] = `
<div>
  hello, 
  , 
</div>
`;

exports[`TestComponent snapshots 0 4`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots 0 5`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots 1 1`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots 1 2`] = `
<div>
  hello, 
  A
  , 
  B
</div>
`;

exports[`TestComponent snapshots 1 3`] = `
<div>
  hello, 
  A
  , 
  B
</div>
`;

exports[`TestComponent snapshots 2 1`] = `
<div>
  hello, 
  A
  , 
  B
</div>
`;

exports[`TestComponent snapshots does thing 1`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots does thing 2`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots does thing 3`] = `
<div>
  hello, 
  A
  , 
</div>
`;

exports[`TestComponent snapshots does things 1`] = `
<div>
  hello, 
  A
  , 
  B
</div>
`;

exports[`TestComponent snapshots does things 2`] = `
<div>
  hello, 
  A
  , 
  B
</div>
`;

exports[`TestComponent snapshots does things 3`] = `
<div>
  hello, 
  A
  , 
  B
</div>
`;

exports[`TestComponent snapshots fake props 1`] = `
<div>
  hello, 
  one
  , 
</div>
`;

exports[`TestComponent snapshots fake props 2`] = `
<div>
  hello, 
  one
  , 
</div>
`;

exports[`TestComponent snapshots fake props 3`] = `
<div>
  hello, 
  one
  , 
</div>
`;

exports[`TestComponent snapshots fake props 4`] = `
<div>
  hello, 
  one
  , 
</div>
`;

exports[`TestComponent snapshots fake props 5`] = `
<TestComponent
  one="one"
>
  <div>
    hello, 
    one
    , 
  </div>
</TestComponent>
`;

exports[`TestComponent snapshots fake props 6`] = `
<div>
  hello, 
  one
  , 
  two
</div>
`;

exports[`TestComponent snapshots with none 1`] = `
<div>
  hello, 
  , 
</div>
`;
